src/helpers/shareRoom.test.ts

```diff
@@ -0,0 +1,135 @@
+                        writeText: mockClipboard,
+                    },
+                });
+
+                await shareRoom();
+
+                expect(mockConsoleError).toHaveBeenCalledWith(`Copy failed with error: Error`);
+            })
+        })
+
+        describe("When the navigator.clipboard.writeText method resolves", () => {
+
+            beforeEach(async () => {
+                const mockClipboard = jest.fn().mockResolvedValueOnce("Success");
+
+                Object.assign(navigator, {
+                    clipboard: {
+                        writeText: mockClipboard,
+                    },
+                });
+            });
+
+            afterEach(() => {
+                document.body.innerHTML = "";
+            });
+
+            it("should append a span to the document body", async () => {
+                await shareRoom();
+
+                const span = document.getElementById('copiedUrlAlert');
+
+                expect(span).not.toBeNull();
+            });
+
+            it("should be appended to the document body", async () => {
+                await shareRoom();
+
+                const span = document.getElementById('copiedUrlAlert') as HTMLSpanElement;
+
+                expect(span.parentElement).toBe(document.body);
+            });
+
+            it("should have the correct innerText", async () => {
+                await shareRoom();
+
+                const span = document.getElementById('copiedUrlAlert') as HTMLSpanElement;
+
+                expect(span.innerText).toBe("URL Copied to Clipboard!");
+            });
+
+            it("should remove the span after 500ms", async () => {
+                jest.useFakeTimers();
+                
+                await shareRoom();
+
+                const span = document.getElementById('copiedUrlAlert') as HTMLSpanElement;
+
+                expect(span).not.toBeNull();
+
+                jest.advanceTimersByTime(500);
+
+                expect(span.parentElement).toBeNull();
+            });
+
+            describe('When the span already exists', () => {
+
+                let span = document.getElementById('copiedUrlAlert') as HTMLSpanElement;
+
+                beforeEach(() => {
+                    span = document.createElement('span')
+                    span.id = 'copiedUrlAlert'
+                    span.innerText = 'URL Copied to Clipboard'
+
+                    document.body.appendChild(span)
+                });
+
+                afterEach(() => {
+                    document.body.removeChild(span)
+                })
+
+                it('Should not append more spans', async () => {
+                    //jest.useFakeTimers(); - is this needed
+
+                    await shareRoom();
+                    await shareRoom();
+                    await shareRoom();
+
+                    const spans = document.querySelectorAll('#copiedUrlAlert');
+
+                    //jest.advanceTimersByTime(500); - is this needed

```

Comment:
You won't need the fake timers here, but you might need it in the other test where it removes the span after the timer